tests/repo_finder/test_provenance_extractor.py

```diff
@@ -0,0 +1,231 @@
+# Copyright (c) 2024 - 2024, Oracle and/or its affiliates. All rights reserved.
```

Comment:
I think it would be great if we can add some unit test cases here for invalid cases:
- There is no repository url.
- There is a repository url but the commit cannot be found.
- Both repository url and commit cannot be found.

Optional, for the `_json_extract` function, we could make use of unit test cases for error cases (where currently exceptions are raised).